libraries/stdlib/src/kotlin/text/Strings.kt

```diff
@@ -1162,3 +1162,84 @@ public fun CharSequence.lineSequence(): Sequence<String> = splitToSequence("\r\n
  * * Splits this char sequence to a list of lines delimited by any of the following character sequences: CRLF, LF or CR.
  */
 public fun CharSequence.lines(): List<String> = lineSequence().toList()
+
+/**
+ * For each i in [0, length), this function computes
+ * the length of the longest suffix of a substring of pattern from 0 to i
+ * that is also a prefix of the pattern itself.
+ */
+private fun computePrefixFunction(pattern: CharSequence): IntArray {
+    val resultTable = IntArray(pattern.length)
+
+    var matches = 0
+    for (i in 1..pattern.length - 1) {
+        while (matches > 0 && pattern[matches] != pattern[i]) {
+            matches = resultTable[matches]
+        }
+
+        if (pattern[matches] == pattern[i]) {
+            matches++
+        }
+        resultTable[i] = matches
+    }
+
+    return resultTable
+}
+
+/**
+ * Returns a list of indices where the pattern occurs in this String. This method
+ * searches character by character and thus does not support regular expressions
+ * as input for the pattern.
+ * For a pattern longer than the text, an empty sequence is returned. If the pattern
+ * is empty, all indices are matched.
+ *
+ * @param [pattern] The pattern to look for in this String. Regular expressions
+ *                 are not supported
+ * @param [ignoreCase] If true, characters are matched even if one is upper and the other is
+ *                 lower case
+ * @param [matchOverlapping] If true, also match overlapping occurences.
+ *
+ * @return A list of indices where the supplied [pattern] starts in the text.
+ */
+public fun CharSequence.occurrencesOf(pattern: CharSequence, ignoreCase: Boolean = false, matchOverlapping: Boolean = false): Sequence<Int> {
+
+    if (pattern.length > this.length) {
+        return emptySequence()
+    }
+
+    if (pattern.isEmpty()) {
+        return indices.asSequence()
```

Comment:
It should be `(0..length).asSequence()`, because you need to count an empty match before the first character and empty one after the last characater.